### ISSUE TITLE
Segmentation fault when using inductor

### ISSUE THREAD

**Issue by ymwangg** [pytorch/torchdynamo#1418]
Minimum code to reproduce:

```py
import torch
import torchdynamo
import torchinductor.config
import logging
from torch import nn

torchinductor.config.debug = True
torchdynamo.config.log_level=logging.DEBUG

class SimpleModel(nn.Module):
    def __init__(self, in_feat=10, out_feat=5):
        super().__init__()

        # segfault
        self.linear = nn.Sequential(
            *[nn.Linear(in_feat, in_feat), nn.ReLU(), nn.Linear(in_feat, out_feat), nn.ReLU()]
        )

    def forward(self, inputs):
        return self.linear(inputs)

device = torch.device("cuda:0")
x = torch.rand(10,10).to(device)

model = SimpleModel().to(device)

model = torchdynamo.optimize("inductor")(model)

y = model(x)
print(y)
```

I tried a couple of changes that can make it work:
1. Disable autograd by adding `with torch.no_grad()` when calling the model.
2. Use other backends such as `aot_nvfuser`.
3. Modify the sequence of operations to be `[nn.Linear(in_feat, in_feat), nn.ReLU(), nn.Linear(in_feat, out_feat)]`, `[nn.Linear(in_feat, in_feat), nn.ReLU(), nn.RELU()]`.

So I suspect this may be related to the `mask` variable with `torch.bool` type saved for backward pass in `nn.ReLU()`.

**Comment by voznesenskym** [comment]
This runs fine for me, no segfault. Where did you see the segfault? Just running the script above? Or running the generated kernel(s)?

**Comment by ymwangg** [comment]
I noticed when I was initially running the script, I got a warning from torchvision.
```
/home/ubuntu/src/pytorch/vision/build/lib.linux-x86_64-3.8/torchvision/io/image.py:13: UserWarning: Failed to load image Python extension: /home/ubuntu/src/pytorch/vision/build/lib.linux-x86_64-3.8/torchvision/image.so: undefined symbol: _ZN2at4_ops19empty_memory_format4callEN3c1014SymIntArrayRefENS2_8optionalINS2_10ScalarTypeEEENS4_INS2_6LayoutEEENS4_INS2_6DeviceEEENS4_IbEENS4_INS2_12MemoryFormatEEE
  warn(f"Failed to load image Python extension: {e}")
```
After removing the stale torchvision, everything seems to work.

Thanks!
